.github/workflows/build_and_push_docker_images.yml

```diff
@@ -0,0 +1,43 @@
+name: Docker build and push
+
+on:
+  pull_request:
+    paths:
+      - core/api/**
+  push:
+    branches: [ develop, main ]
+    paths:
+      - core/api/**
+  workflow_dispatch:
+
+jobs:
+  docker_build:
+    name: Build management-api
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Login to Packages Container registry
+        uses: docker/login-action@v1 
+        with:
+          registry: docker.pkg.github.com
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build & publish image
+        env:
+          DOCKER_BUILDKIT: 1
+          REPOSITORY_NAME: docker.pkg.github.com/microsoft/azuretre/management-api
+        run: |
+          echo $GITHUB_REF
+          if [ $GITHUB_EVENT_NAME == 'push' ] && [ $GITHUB_REF == 'refs/heads/develop' ]; then
+            TAG='develop-latest'
+          elif [ $GITHUB_EVENT_NAME == 'push' ] && [ $GITHUB_REF == 'refs/heads/main' ]; then
+            TAG='main-latest'
+          else
+            TAG=$(jq --raw-output .pull_request.number $GITHUB_EVENT_PATH)
```

Comment:
Does this differentiate image builds on a given PR or should we include a git sha or index? Just thinking there might be multiple images built for a PR given feedback and iteration etc. and it'll be better to be able to pin point which image refers to which iteration and also ensure that any PR test infra picks up a specific image.

Reply:
No, all images in the PR are tagged the same, so overwrites. True, need to avoid caching and trigger a change. Will change to git sha.